Rakefile

```diff
@@ -31,16 +31,96 @@ task :test, [:name] do |name, args|
     puts "********** Running test #{test_name} **********"
     puts "---------------------------------------------------------------------------------"
 
-    puts "\n****** Creating KVM image ******\n"
+    testing_iso = File.join(base_dir, "kiwi/iso/testing.iso")
+    obs_iso = File.join(base_dir, "kiwi/iso/obs.iso")
     autoyast_file = File.join(base_dir, "kiwi/definitions/autoyast/autoinst.xml")
-    FileUtils.cp( File.join(base_dir, "spec", test_name + ".xml"), autoyast_file)
+    dest_definition = File.join(base_dir, "kiwi/definitions/autoyast/definition.rb")
+    default_iso = "http://dist.suse.de/install/SLE-12-Server-GM/SLE-12-Server-DVD-x86_64-GM-DVD1.iso"
+
+    FileUtils.rm(testing_iso, :force => true)
+
+    # Set used iso
+    unless test_name.start_with?("upgrade_")
+      # New installation workflow. So we are take the built OBS iso
+      FileUtils.ln(obs_iso, testing_iso) if File.file?(obs_iso)
+    end
+    # Set download iso (if it is not already available)
+    src_definition = File.join(base_dir, "kiwi/definitions/autoyast/install_definition.rb")
+    FileUtils.cp(src_definition, dest_definition)
+    iso_path_file = File.join(base_dir, "spec", test_name + ".install_iso")
+    if File.file?(iso_path_file)
+      data = IO.binread(iso_path_file).chomp
+    else
+      data = default_iso
+    end
+    system "sed -i.bak s,__iso_source_path__,#{data},g #{dest_definition}"
+
+    # Set autoinst.xml
+    autoinst = File.join(base_dir, "spec", test_name + ".xml")
+    autoinst = File.join(base_dir, "spec", test_name + ".install_xml") unless File.file?(autoinst)
+    FileUtils.cp( autoinst, autoyast_file)
+
+    puts "\n****** Creating KVM image ******\n"
     Dir.chdir(File.join( base_dir, "kiwi")) {
       puts "\n**** Building KVM image ****\n" 
-      system "veewee kvm build autoyast --force"
+      system "veewee kvm build autoyast --force --auto"
+    }
+
+    if test_name.start_with?("upgrade_")
+      # upgrade workflow
+      # Take the generated image and run a autoyast upgrade workflow
+
+      # Set used iso
+      FileUtils.rm(testing_iso, :force => true)
+      FileUtils.ln(obs_iso, testing_iso) if File.file?(obs_iso) #Taking obs iso for upgrade
+
+      # Change boot order to CD on the top
+      system "sudo virsh destroy autoyast"
+      system "sudo virsh dumpxml autoyast >autoyast_description.xml"
+      system "sudo sed -i.bak s/dev=\\'cdrom\\'/dev=\\'cdrom_save\\'/g autoyast_description.xml"
```

Comment:
Maybe I'm missing some detail, but I'd say that it should be enough changing `boot dev='hd'` for `boot dev='cdrom'`. Or not?